gef.py

```diff
@@ -6764,18 +6764,13 @@ def do_invoke(self, argv):
             tids = [current_thread.num]
 
         # As a nicety, we want to display threads in ascending order by gdb number
-        for thread in sorted(threads, key=lambda t: t.num):
+        for thread in threads:
             if thread.num not in tids:
                 continue
 
             thread.switch()
-            try:
-                tcache_addr = gdb.parse_and_eval("(void *) tcache")
-            except gdb.error:
-                # In binaries not linked with pthread, we can't use the tcache
-                # symbol.
-                tcache_addr = self.get_main_thread_tcache()
 
```

Comment:
There should be a test for libc version: there's no need to pursue further down if libc < (2, 26) - i.e. buffer tcache support

Reply:
You don't think the check at the beginning of `do_invoke` is adequate? Or are we concerned about this function being called from a different location?

Reply:
Hum, probably not now that I think of it. Forget what I said 😄